packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -351,7 +351,7 @@ export class Secret extends SecretBase {
     });
 
     this.encryptionKey = props.encryptionKey;
-    this.secretName = this.physicalName;
+    this.secretName = props.secretName ?? parseSecretName(this, this.secretArn);
```

Comment:
I feel we need a way to return a token that encodes the implicit dependency. Otherwise this might not behave as expected. 

If some resource references `secret.secretName` in its CFN props, there's an expectation that this will resolve in CFN as a Ref/GetAtt and then an implicit dependency will be formed.

Reply:
I'm not quite sure I understand. Do you mean even the `props.secretName` case should be tokenized? We could certainly just always use `parseSecretName`. The downside would be this:

```ts
const secretName = 'SuperImportant';
const secret = new Secret(this, 'Secret', { secretName });
secret.secretName === secretName // false
```

Not the end of the world perhaps, but slightly unexpected.

---

Sidenote - I'm not treating this as a breaking change, because I'm viewing the use case here (of referencing the `secretName` post-creation) as broken for all users who haven't explicitly set the secretName in the props; those users would have a Token that would never resolve and so wouldn't have been able to accomplish anything with it. Opting for always tokenizing does run the slightest risk of breaking something for someone who _has_ supplied the secret name and is using it. This would be pretty esoteric (e.g., later looking up secrets in a hashmap by their secret name or something), but I figure it's worth calling out.

Reply:
yes, as far as I know this is the approach we've been taking across the library to ensure implicit dependencies are recorded in the CFN template.